diff --git a/core/coretypes/include/coretypes/bb_exception.h b/core/coretypes/include/coretypes/bb_exception.h
index 8fd8603..121b81a 100644
--- a/core/coretypes/include/coretypes/bb_exception.h
+++ b/core/coretypes/include/coretypes/bb_exception.h
@@ -25,7 +25,7 @@ class DaqException : public std::runtime_error
 public:
     template <typename... Params>
     explicit DaqException(ErrCode errCode, const std::string& format, Params&&... params)
-        : DaqException(false, errCode, fmt::format(format, std::forward<Params>(params)...))
+        : DaqException(false, errCode, fmt::format(fmt::runtime(format), std::forward<Params>(params)...))
     {
     }
 
