diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index aa5b454..a85557d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -178,30 +178,16 @@ if (GGML_OPENBLAS)
 endif()
 
 if (GGML_CLBLAST)
-    if(NOT CLBLAST_HOME)
-        set(CLBLAST_HOME
-            /usr
-            /usr/local
-            $ENV{CLBLAST_HOME}
-            )
-    endif()
-
-    find_path(CLBLAST_INC NAMES clblast.h PATHS ${CLBLAST_HOME} PATH_SUFFIXES include)
-    find_library(CLBLAST_LIB NAMES clblast PATHS ${CLBLAST_HOME} PATH_SUFFIXES build)
-    find_path(OPENCL_INC NAMES CL/opencl.h PATHS ${OPENCL_ROOT})
-    find_library(OPENCL_LIB NAMES OpenCL)
-    if (CLBLAST_LIB AND OPENCL_LIB AND CLBLAST_INC AND OPENCL_INC)
-        message(STATUS "clBLAST found")
-
-        set(GGML_EXTRA_INCS  ${GGML_EXTRA_INCS}  ${CLBLAST_INC} ${OPENCL_INC})
-        set(GGML_EXTRA_LIBS  ${GGML_EXTRA_LIBS}  ${CLBLAST_LIB} ${OPENCL_LIB})
-        set(GGML_EXTRA_FLAGS ${GGML_EXTRA_FLAGS} -DGGML_USE_CLBLAST)
+    find_package(CLBlast REQUIRED)
+    if (CLBlast_FOUND)
+        message(STATUS "CLBlast found")
 
         set(GGML_OPENCL_SOURCES ggml-opencl.cpp ggml-opencl.h)
 
-        link_libraries("-Wl,--copy-dt-needed-entries")
+        add_compile_definitions(GGML_USE_CLBLAST)
+        set(GGML_EXTRA_LIBS ${LLAMA_EXTRA_LIBS} clblast)
     else()
-        message(WARNING "clBLAST not found")
+        message(WARNING "CLBlast not found")
     endif()
 endif()
 
@@ -213,7 +199,7 @@ endif()
 if (GGML_CUDA)
     cmake_minimum_required(VERSION 3.17)
 
-    find_package(CUDAToolkit)
+    find_package(CUDAToolkit REQUIRED)
     if (CUDAToolkit_FOUND)
         message(STATUS "CUDA found")
 
